paddle/fluid/operators/reshape_op.cc

```diff
@@ -56,6 +56,9 @@ class ReshapeOp : public framework::OperatorWithKernel {
   static framework::DDim ValidateShape(const std::vector<int> shape,
                                        const framework::DDim &in_dims) {
     const int64_t in_size = framework::product(in_dims);
+    auto in_dims_vec = framework::vectorize(in_dims);
+    bool all_positive = std::all_of(in_dims_vec.cbegin(), in_dims_vec.cend(),
+                                    [](int64_t i) { return i > 0; });
```

Comment:
If the `in_dims` has two negative value, the result of `framework::product(in_dims)` will be positive, it is not consistent with line94.

Reply:
do we have a test for this?

Reply:
Currently, we don't have a test for this.
There is an example: when the input's dim is (8, -1, -1,) and the `shape` is (8, -1), the output's dim should be (8, -1), but the origin implementation will infer the outout'dim to (8, 1).